### ISSUE TITLE
Loki metadata available at variable's querying 

### ISSUE THREAD

**Issue by itepifanio** [grafana/grafana#84909]
**Why is this needed**:

Loki metadata provides more flexibility to design query without worrying about cardinality problems. To fully enable loki's powers to create dashboards with Grafana the metadata querying should be also available at variables.

**What would you like to be added**:

I'm currently migrating our stack from elastic to grafana. Some of the filters available on elastic's dashboard can't be used at Loki due to the lack of metadata variable's querying feature.

**Who is this feature for?**

All Grafana users that uses the newest versions of Loki as data source

**Comment by mcaxtrzen** [comment]
any updates?

**Comment by PabloEForgeFlow** [comment]
This would be quite handy to query structured metadata generated when converting [OpenTelemetry logs](https://grafana.com/docs/grafana-cloud/send-data/otlp/otlp-format-considerations/#logs) to Loki.

As far as I can tell `detected_level` is not a real label, so it can't be queried. Being able to acess `metadata[severity_text]` would be nice.

**Comment by dccarbone** [comment]
Would be amazing to have that! 🙏

**Comment by pschiffe** [comment]
Also just stumbled on this, that you currently cannot query structured metadata as variables. Grafana's monitoring helm chart recently switched pod names in logs from labels to metadata, but that breaks dashboards - https://github.com/grafana/k8s-monitoring-helm/pull/1542

**Comment by adrianrademann** [comment]
We ran into the same issue when collecting frontend telemetry via the opentelemetry faro receiver.

**Comment by MarijnMB** [comment]
For anyone looking for a workaround: I'm settings labels in the alloy collector from the metadata. Make sure to use different naming.

**Comment by smoebody** [comment]
> For anyone looking for a workaround: I'm settings labels in the alloy collector from the metadata. Make sure to use different naming.

This is exactly what you want to prevent with _structured_metadata_. As I understand Loki each combination of label values results in a separate log stream with its own chunk and index files. Lets say you have three labels with 5 different values each you will end up in `5*5*5` = 100 log streams.

**Comment by MarijnMB** [comment]
> This is exactly what you want to prevent with _structured_metadata_. As I understand Loki each combination of label values results in a separate log stream with its own chunk and index files. Lets say you have three labels with 5 different values each you will end up in `5*5*5` = 100 log streams.

If you have a solution or better idea/workaround, I'm all ears.

**Comment by smoebody** [comment]
> > This is exactly what you want to prevent with _structured_metadata_. As I understand Loki each combination of label values results in a separate log stream with its own chunk and index files. Lets say you have three labels with 5 different values each you will end up in `5*5*5` = 100 log streams.
> 
> If you have a solution or better idea/workaround, I'm all ears.

No, I came here in search for a solution myself. I just wanted to mention the tradeoffs of your suggested workaround.
